### ISSUE TITLE
Drupal Rules and DrupalGap Push Notifications

### ISSUE THREAD

**Issue by mkinnan** [signalpoint/push_notifications#15]
Integrating push notifications with rules is now possible. I've tested and it works. Rules is configured to send just a message (_no option for title, sound, etc._). Should rules be where the title, message, count, etc. gets configured? Or does **hook_push_notifications_receive()** override any message sent by the push notification module to drupalgap?

**Comment by signalpoint** [comment]
@mkinnan Yes, the push notifications module up in Drupal is where the magic needs to happen. It appears they don't allow for sending anything other than the message. It'd be awesome to be able to specify those extra `data` properties.

The `hook_push_notifications_receive()` in the DrupalGap SDK is where the developer decides what to do with the notification once it is received on the app. It used to auto alert the push notification, but I figure (and a client already needed) a way to customize what happens when the notification is received.

Either way, at this point I've only successfully received a message, none of those other `data` options provided by the PhoneGap/Cordova plugin. I'd check out the issue queue on d.o for push_notifications to see what progress/issues are out there in regards to improving the flexibility on the Drupal side.

**Comment by mkinnan** [comment]
I am looking into the push notifications module for drupal. Putting some links here for reference later:

http://stackoverflow.com/questions/21666298/android-push-notifications-how-do-i-show-a-title

https://developers.google.com/cloud-messaging/http-server-ref#notification-payload-support

https://gist.github.com/prime31/5675017

http://stackoverflow.com/questions/32225208/how-to-send-alert-as-array-in-ios-push-notifications-in-php

It looks like the push notifications module needs to be split up more in the code to do more in terms of title, message, sound, etc between iOS and Android. It appears this is a default for both iOS and Android

`$payload = array('alert' => $message);`

**Comment by mkinnan** [comment]
![screenshot_2016-03-17-19-56-01-1](https://cloud.githubusercontent.com/assets/13283396/13866409/a2ee3d10-ec7a-11e5-8cb1-052b0232c69c.png)

**Comment by signalpoint** [comment]
Do tell...
- Tyler (sent from mobile device)

**Comment by mkinnan** [comment]
Made some changes to the d.o push notifications module ... just the **admin/config/services/push_notifications/message** page for right now. I'll tinker and try to add option for icons and sounds. Once I get that figured out, will try to get it working on rules. Then make a patch and post on the d.o issue queue.

**Comment by mkinnan** [comment]
Title functionality added and PHP evaluation for rules:
https://www.drupal.org/node/2686101#comment-10985941
